app/endpoints/notification.py

```diff
@@ -0,0 +1,256 @@
+    )
+
+    if firebase_device is not None:
+        if firebase_device.user_id == user.id:
+            # Update the register date
+            await cruds_notification.update_firebase_devices_register_date(
+                firebase_device_token=firebase_token,
+                new_register_date=date.today(),
+                db=db,
+            )
+            return
+        # If the user is not the same, we don't want the new user to receive the notifications of the old one
+        await cruds_notification.delete_firebase_devices(
+            firebase_device_token=firebase_token, db=db
+        )
+
+    # We also need to subscribe the new token to the topics the user is subscribed to
+    topic_memberships = await cruds_notification.get_topic_membership_by_user_id(
+        user_id=user.id, db=db
+    )
+
+    for topic_membership in topic_memberships:
+        await notification_manager.subscribe_tokens_to_topic(
+            tokens=[firebase_token], topic=topic_membership.topic
+        )
+
+    firebase_device = models_notification.FirebaseDevice(
+        user_id=user.id,
+        firebase_device_token=firebase_token,
+        register_date=date.today(),
+    )
+
+    await cruds_notification.create_firebase_devices(
+        firebase_device=firebase_device, db=db
+    )
+
+
+@router.delete(
+    "/notification/devices/{firebase_token}",
+    status_code=204,
+    tags=[Tags.notifications],
+)
+async def unregister_firebase_device(
+    firebase_token: str,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+    notification_manager: NotificationManager = Depends(get_notification_manager),
+):
+    """
+    Unregister a new firebase device for the user
+
+    **The user must be authenticated to use this endpoint**
+    """
+    # Anybody may unregister a device if they know its token, which should be secret
+
+    # We also need to unsubscribe the token to the topics the user is subscribed to
+    topic_memberships = await cruds_notification.get_topic_membership_by_user_id(
+        user_id=user.id, db=db
+    )
+
+    for topic_membership in topic_memberships:
+        await notification_manager.unsubscribe_tokens_to_topic(
+            tokens=[firebase_token], topic=topic_membership.topic
+        )
+
+    await cruds_notification.delete_firebase_devices(
+        firebase_device_token=firebase_token, db=db
+    )
+
+
+@router.get(
+    "/notification/messages/{firebase_token}",
+    response_model=list[schemas_notification.Message],
+    status_code=200,
+    tags=[Tags.notifications],
+)
+async def get_messages(
+    firebase_token: str,
+    db: AsyncSession = Depends(get_db),
+    #    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    """
+    Get all messages for a specific device from the user
+
+    **The user must be authenticated to use this endpoint**
+    """
+    # TODO: enable authentification for this endpoint
```

Comment:
It depends if Titan can fetch Messages after the authorization process. @maximeroucher do you know if this would be possible?

Reply:
It is not possible currently because of background notification, which are handled in a separate VM, which does not have access to Titan, to make it work with auth, it will require to reimplement the whole login system, which can maybe be done, but will be very tedious, it could be however be a Titan issue for now